frontend/src/pages/CategoryPage.tsx

```diff
@@ -23,29 +20,32 @@ export default function CategoryPage() {
 
     const addCategory: FormEventHandler<HTMLFormElement> = (event) => {
         event.preventDefault()
-        const form = event.currentTarget
-        const formElements = form.elements as typeof form.elements & ITextInput
-        const categoryName = formElements.categoryInput.value.trim()
+        const categoryName: string = categoryInput.trim()
 
         if (categoryName && categoryName.length > 0) {
             controller.addCategory(categoryName).then(setCategories)
         }
-        formElements.categoryInput.value = "";
+
+        // @ts-ignore
+        event.currentTarget.elements[0].value = ""
```

Comment:
Add the value prop to the input. Then the input is fully controlled and you can reset it via the state... :)